sentry-ruby/lib/sentry/metrics/aggregator.rb

```diff
@@ -0,0 +1,185 @@
+# frozen_string_literal: true
+
+module Sentry
+  module Metrics
+    class Aggregator
+      include LoggingHelper
+
+      FLUSH_INTERVAL = 5
+      ROLLUP_IN_SECONDS = 10
+
+      KEY_SANITIZATION_REGEX = /[^a-zA-Z0-9_\/.-]+/
+      VALUE_SANITIZATION_REGEX = /[^[[:word:]][[:digit:]][[:space:]]_:\/@\.{}\[\]$-]+/
+
+      METRIC_TYPES = {
+        c: CounterMetric,
+        d: DistributionMetric,
+        g: GaugeMetric,
+        s: SetMetric
+      }
+
+      # exposed only for testing
+      attr_reader :thread, :buckets, :flush_shift
+
+      def initialize(configuration, client)
+        @client = client
+        @logger = configuration.logger
+
+        @default_tags = {}
+        @default_tags['release'] = configuration.release if configuration.release
+        @default_tags['environment'] = configuration.environment if configuration.environment
```

Comment:
Do we have the fallback to `production` somewhere?

Reply:
yes, but it's not production
https://github.com/getsentry/sentry-ruby/blob/c1217cd450adb5343ed287aa92b8a6d45c5bb91d/sentry-ruby/lib/sentry/configuration.rb#L637-L639